### ISSUE TITLE
[rawhide] kola ISO tests failing with kernel null pointer dereference bugs

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#898]
If you try to boot an ISO with [kernel-5.14.0-0.rc1.16.fc35](https://bodhi.fedoraproject.org/updates/FEDORA-2021-ba77f1a335) and later for some reason the system locks up and spews null pointer de-reference traces.

```
[    8.849896] BUG: kernel NULL pointer dereference, address: 0000000000000010
[    8.850941] #PF: supervisor read access in kernel mode
[    8.851691] #PF: error_code(0x0000) - not-present page
[    8.852422] PGD 0 P4D 0 
[    8.852803] Oops: 0000 [#1] SMP NOPTI
[    8.853321] CPU: 1 PID: 20 Comm: kworker/1:0 Tainted: G        W        --------- ---  5.14.0-0.rc1.16.fc35.x86_64 #1
[    8.854790] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.14.0-1.fc33 04/01/2014
[    8.856053] Workqueue: events ttm_device_delayed_workqueue [ttm]
[    8.856944] RIP: 0010:qxl_bo_delete_mem_notify+0x19/0x40 [qxl]
[    8.857793] Code: 89 e7 45 31 e4 e8 b7 1c cb ef eb ea 0f 1f 44 00 00 0f 1f 44 00 00 55 48 89 fd e8 a2 02 00 00 84 c0 74 0d 48 8b 85 68 01 00 00 <83> 78 10 03 74 02 5d c3 8b 85 64 02 00 00 85 c0 74 f4 48 8b 7d 08
[    8.860522] RSP: 0018:ffffa5b4400b3dd0 EFLAGS: 00010202
[    8.861287] RAX: 0000000000000000 RBX: ffff8b86c1822670 RCX: ffff8b86c033cf80
[    8.862319] RDX: ffff8b86ce2eb990 RSI: ffff8b86ce2eb990 RDI: ffff8b86ce2eb800
[    8.863360] RBP: ffff8b86ce2eb800 R08: ffff8b86ce2eb990 R09: 0000000000000000
[    8.864397] R10: 0000000000000001 R11: 0000000000000000 R12: ffff8b86ce2eb990
[    8.865438] R13: ffff8b86ce2eb800 R14: ffff8b86ce2eb960 R15: ffff8b86ce2eb990
[    8.866483] FS:  0000000000000000(0000) GS:ffff8b86f8d00000(0000) knlGS:0000000000000000
[    8.867649] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    8.868481] CR2: 0000000000000010 CR3: 0000000111c28000 CR4: 0000000000350ee0
[    8.869515] Call Trace:
[    8.869871]  ttm_bo_cleanup_memtype_use+0x22/0x60 [ttm]
[    8.870648]  ttm_bo_release+0x1a1/0x300 [ttm]
[    8.871294]  ttm_bo_delayed_delete+0x1be/0x220 [ttm]
[    8.872027]  ttm_device_delayed_workqueue+0x18/0x40 [ttm]
[    8.872831]  process_one_work+0x1ec/0x390
[    8.873430]  worker_thread+0x53/0x3e0
[    8.873978]  ? process_one_work+0x390/0x390
[    8.874598]  kthread+0x127/0x150
[    8.875073]  ? set_kthread_struct+0x40/0x40
[    8.875692]  ret_from_fork+0x22/0x30
[    8.876229] Modules linked in: rfkill snd_hda_codec_generic ledtrig_audio intel_rapl_msr snd_hda_intel snd_intel_dspcfg snd_intel_sdw_acpi snd_hda_codec qxl snd_hda_core drm_ttm_helper ttm snd_hwdep snd_pcsp drm_kms_helper intel_rapl_common snd_pcm snd_timer virtio_balloon joydev snd soundcore cec i2c_piix4 drm ip_tables squashfs nls_utf8 isofs dm_multipath crct10dif_pclmul crc32_pclmul crc32c_intel ghash_clmulni_intel serio_raw virtio_console e1000 ata_generic pata_acpi qemu_fw_cfg xfs loop fuse
[    8.882535] CR2: 0000000000000010
[    8.883019] ---[ end trace c31395b6993f2691 ]---
[    8.883703] RIP: 0010:qxl_bo_delete_mem_notify+0x19/0x40 [qxl]
[    8.884539] Code: 89 e7 45 31 e4 e8 b7 1c cb ef eb ea 0f 1f 44 00 00 0f 1f 44 00 00 55 48 89 fd e8 a2 02 00 00 84 c0 74 0d 48 8b 85 68 01 00 00 <83> 78 10 03 74 02 5d c3 8b 85 64 02 00 00 85 c0 74 f4 48 8b 7d 08
[    8.887185] RSP: 0018:ffffa5b4400b3dd0 EFLAGS: 00010202
[    8.887936] RAX: 0000000000000000 RBX: ffff8b86c1822670 RCX: ffff8b86c033cf80
[    8.888946] RDX: ffff8b86ce2eb990 RSI: ffff8b86ce2eb990 RDI: ffff8b86ce2eb800
[    8.889960] RBP: ffff8b86ce2eb800 R08: ffff8b86ce2eb990 R09: 0000000000000000
[    8.890979] R10: 0000000000000001 R11: 0000000000000000 R12: ffff8b86ce2eb990
[    8.892022] R13: ffff8b86ce2eb800 R14: ffff8b86ce2eb960 R15: ffff8b86ce2eb990
[    8.893042] FS:  0000000000000000(0000) GS:ffff8b86f8d00000(0000) knlGS:0000000000000000
[    8.894207] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    8.895137] CR2: 0000000000000010 CR3: 0000000111c28000 CR4: 0000000000350ee0
```

Talked to jforbes. He's aware and says it's fixed upstream:

```
15:57:12 dustymabe | hey jforbes - are null pointer dereference bugs something
                   | you're interested in? https://paste.centos.org/view/62754818
15:58:34   jforbes | That one is known and being dealt with upstream
15:59:11   jforbes | Fixed by 3efe180d5105d367ae1dfadb97892ab93a89a783 drm/qxl: add
                   | NULL check for bo->resource
15:59:46   jforbes | Ah, which hasn't been pulled yet, that is the tree fix, but it
                   | is staged
16:00:12 dustymabe | jforbes: +1 
```

Let's pin the kernel in rawhide for now to get around this.

**Comment by dustymabe** [comment]
still an issue in [kernel-5.14.0-0.rc1.20210714git40226a3d96ef.18.fc35](https://bodhi.fedoraproject.org/updates/FEDORA-2021-9e3da87521). Will stay pinned for now.

**Comment by jlebon** [comment]
CI in https://github.com/coreos/fedora-coreos-config/pull/1110 is green now, and it pulled in that same kernel, but I'm guessing that's part of the flakiness?

**Comment by dustymabe** [comment]
yeah - it's intermittent. The test I'm doing right now is a local build and just booting the live ISO. 9/10 times i never even make it to a bash prompt before getting a kernel trace for a null pointer dereference.

**Comment by dustymabe** [comment]
This now seems to be fixed with [kernel-5.14.0-0.rc2.23.fc35](https://bodhi.fedoraproject.org/updates/FEDORA-2021-4d98455de6).

**Comment by dustymabe** [comment]
Closing this now that https://github.com/coreos/fedora-coreos-config/pull/1110 has merged.
